lib/rules/prefer-object-has-own.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * @fileoverview Prefers Object.hasOwn() instead of Object.prototype.hasOwnProperty.call()
+ * @author Nitin Kumar
+ * @author Gautam Arora
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const astUtils = require("./utils/ast-utils");
+
+/**
+ * Checks to see if a property name object exists in the subtree recursively.
+ * @param {ASTnode} node to evalutate.
+ * @returns {boolean} `true` if object property exists, `false` otherwise.
+ */
+function hasLeftHandObject(node) {
+    if (!node.object) {
+        return false;
+    }
+
+    const objectPropertyName = astUtils.getStaticPropertyName(node.object);
+    let objectNodeToCheck = node.object;
+
+    if (objectPropertyName === "prototype") {
+        objectNodeToCheck = node.object.object;
+
+        // for case - `({}).prototype.hasOwnProperty.call(a, b)`
+        if (objectNodeToCheck.type === "ObjectExpression") {
+            return false;
+        }
```

Comment:
Can we simplify this by checking for `ObjectExpression` only once, separately from `Object` identifier? For example, we can have something like this at the beginning of this function, and after that check only for `Object` identifier: 

```js
if (node.object.type === "ObjectExpression" && node.object.properties.length === 0) {
    return true;
}
```

Reply:
Yes, simplified 👍🏻